@dnceng - AFAICT the "Build Windows_NT x86 release Runtime_Debug" is failing with the error:
```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(231,5): error MSB8065: Custom build for item "F:\workspace\_work\1\s\artifacts\obj\native\netcoreapp5.0-Windows_NT-Release-x86\CMakeFiles\01d9c98fded5d1e41907661a010b2e75\INSTALL_force.rule" succeeded, but specified output "f:\workspace\_work\1\s\artifacts\obj\native\netcoreapp5.0-windows_nt-release-x86\cmakefiles\install_force" has not been created. This may cause incremental build to work incorrectly. [F:\workspace\_work\1\s\artifacts\obj\native\netcoreapp5.0-Windows_NT-Release-x86\install.vcxproj] [F:\workspace\_work\1\s\src\libraries\Native\build-native.proj]
```
Which seems to be a frequent error message, but doesn't seem to usually cause the build to actually fail. I am able to build locally.

--- a/src/coreclr/src/jit/codegen.h
+++ b/src/coreclr/src/jit/codegen.h
@@ -833,6 +833,10 @@ protected:
     void genLongToIntCast(GenTree* treeNode);
 #endif
 
+    // Generate code for a GT_BITCAST that is not contained.
+    void genCodeForBitCast(GenTreeOp* treeNode);
+
+    // Generate the instruction to move a value between register files
     void genBitCast(var_types targetType, regNumber targetReg, var_types srcType, regNumber srcReg);
 
     struct GenIntCastDesc

--- a/src/coreclr/src/jit/codegenarmarch.cpp
+++ b/src/coreclr/src/jit/codegenarmarch.cpp
@@ -255,7 +255,6 @@ void CodeGen::genCodeForTreeNode(GenTree* treeNode)
             {
                 regNumber srcReg = op1->GetRegNum();
                 assert(genTypeSize(op1->TypeGet()) == genTypeSize(targetType));
-                instruction ins = ins_Copy(srcReg, targetType);
 #ifdef TARGET_ARM
                 if (genTypeSize(targetType) == 8)
                 {
@@ -264,7 +263,7 @@ void CodeGen::genCodeForTreeNode(GenTree* treeNode)
                     {
                         regNumber otherReg = treeNode->AsMultiRegOp()->gtOtherReg;
                         assert(otherReg != REG_NA);
-                        inst_RV_RV_RV(INS_vmov_d2i, targetReg, otherReg, genConsumeReg(op1), EA_8BYTE);
+                        inst_RV_RV_RV(INS_vmov_d2i, targetReg, otherReg, srcReg, EA_8BYTE);
                     }
                     else
                     {
@@ -274,6 +273,7 @@ void CodeGen::genCodeForTreeNode(GenTree* treeNode)
                 else
 #endif // TARGET_ARM
                 {
+                    instruction ins = ins_Copy(srcReg, targetType);
                     inst_RV_RV(ins, targetReg, srcReg, targetType);
                 }
             }

--- a/src/coreclr/src/jit/codegenarmarch.cpp
+++ b/src/coreclr/src/jit/codegenarmarch.cpp
@@ -239,50 +239,8 @@ void CodeGen::genCodeForTreeNode(GenTree* treeNode)
             break;
 
         case GT_BITCAST:
-        {
-            GenTree* op1 = treeNode->AsOp()->gtOp1;
-            genConsumeRegs(op1);
-            if (op1->isContained())
-            {
-                assert(op1->IsLocal() || op1->isIndir());
-                op1->gtType = treeNode->TypeGet();
-                op1->SetRegNum(targetReg);
-                op1->ClearContained();
-                JITDUMP("Changing type of BITCAST source to load directly.");
-                genCodeForTreeNode(op1);
-            }
-            else if (varTypeIsFloating(treeNode) != varTypeIsFloating(op1))
-            {
-                regNumber srcReg = op1->GetRegNum();
-                assert(genTypeSize(op1->TypeGet()) == genTypeSize(targetType));
-#ifdef TARGET_ARM
-                if (genTypeSize(targetType) == 8)
-                {
-                    // Converting between long and double on ARM is a special case.
-                    if (targetType == TYP_LONG)
-                    {
-                        regNumber otherReg = treeNode->AsMultiRegOp()->gtOtherReg;
-                        assert(otherReg != REG_NA);
-                        inst_RV_RV_RV(INS_vmov_d2i, targetReg, otherReg, srcReg, EA_8BYTE);
-                    }
-                    else
-                    {
-                        NYI_ARM("Converting from long to double");
-                    }
-                }
-                else
-#endif // TARGET_ARM
-                {
-                    instruction ins = ins_Copy(srcReg, targetType);
-                    inst_RV_RV(ins, targetReg, srcReg, targetType);
-                }
-            }
-            else
-            {
-                inst_RV_RV(ins_Copy(targetType), targetReg, genConsumeReg(op1), targetType);
-            }
-        }
-        break;
+            genCodeForBitCast(treeNode->AsOp());
+            break;
 
         case GT_LCL_FLD_ADDR:
         case GT_LCL_VAR_ADDR:
@@ -1110,6 +1068,59 @@ void CodeGen::genPutArgReg(GenTreeOp* tree)
     }
 
     genProduceReg(tree);
+}
+
+//----------------------------------------------------------------------
+// genCodeForBitCast - Generate code for a GT_BITCAST that is not contained
+//
+// Arguments
+//    treeNode - the GT_BITCAST for which we're generating code
+//
+void CodeGen::genCodeForBitCast(GenTreeOp* treeNode)
+{
+    regNumber targetReg  = treeNode->GetRegNum();
+    var_types targetType = treeNode->TypeGet();
+    GenTree*  op1        = treeNode->gtGetOp1();
+    genConsumeRegs(op1);
+    if (op1->isContained())
+    {
+        assert(op1->IsLocal() || op1->isIndir());
+        op1->gtType = treeNode->TypeGet();
+        op1->SetRegNum(targetReg);
+        op1->ClearContained();
+        JITDUMP("Changing type of BITCAST source to load directly.");
+        genCodeForTreeNode(op1);
+    }
+    else if (varTypeIsFloating(treeNode) != varTypeIsFloating(op1))
+    {
+        regNumber srcReg = op1->GetRegNum();
+        assert(genTypeSize(op1->TypeGet()) == genTypeSize(targetType));
+#ifdef TARGET_ARM
+        if (genTypeSize(targetType) == 8)
+        {
+            // Converting between long and double on ARM is a special case.
+            if (targetType == TYP_LONG)
+            {
+                regNumber otherReg = treeNode->AsMultiRegOp()->gtOtherReg;
+                assert(otherReg != REG_NA);
+                inst_RV_RV_RV(INS_vmov_d2i, targetReg, otherReg, srcReg, EA_8BYTE);
+            }
+            else
+            {
+                NYI_ARM("Converting from long to double");
+            }
+        }
+        else
+#endif // TARGET_ARM
+        {
+            instruction ins = ins_Copy(srcReg, targetType);
+            inst_RV_RV(ins, targetReg, srcReg, targetType);
+        }
+    }
+    else
+    {
+        inst_RV_RV(ins_Copy(targetType), targetReg, genConsumeReg(op1), targetType);
+    }
 }
 
 #if FEATURE_ARG_SPLIT

--- a/src/coreclr/src/jit/codegenxarch.cpp
+++ b/src/coreclr/src/jit/codegenxarch.cpp
@@ -1736,35 +1736,8 @@ void CodeGen::genCodeForTreeNode(GenTree* treeNode)
             break;
 
         case GT_BITCAST:
-        {
-            GenTree* const op1 = treeNode->AsOp()->gtOp1;
-            genConsumeRegs(op1);
-
-            if (op1->isContained())
-            {
-                assert(op1->IsLocal() || op1->isIndir());
-                if (genIsRegCandidateLocal(op1))
-                {
-                    unsigned lclNum = op1->AsLclVar()->GetLclNum();
-                    GetEmitter()->emitIns_R_S(ins_Load(treeNode->TypeGet(), compiler->isSIMDTypeLocalAligned(lclNum)),
-                                              emitTypeSize(treeNode), targetReg, lclNum, 0);
-                }
-                else
-                {
-                    op1->gtType = treeNode->TypeGet();
-                    op1->SetRegNum(targetReg);
-                    op1->ClearContained();
-                    JITDUMP("Changing type of BITCAST source to load directly.");
-                    genCodeForTreeNode(op1);
-                }
-            }
-            else
-            {
-                genBitCast(targetType, targetReg, op1->TypeGet(), op1->GetRegNum());
-            }
-            genProduceReg(treeNode);
+            genCodeForBitCast(treeNode->AsOp());
             break;
-        }
 
         case GT_LCL_FLD_ADDR:
         case GT_LCL_VAR_ADDR:
@@ -7452,7 +7425,7 @@ void CodeGen::genIntrinsic(GenTree* treeNode)
 }
 
 //----------------------------------------------------------------------
-// genBitCast - Generate code for a GT_BITCAST
+// genBitCast - Generate the instruction to move a value between register files
 //
 // Arguments
 //    targetType - the destination type
@@ -7489,6 +7462,44 @@ void CodeGen::genBitCast(var_types targetType, regNumber targetReg, var_types sr
     {
         inst_RV_RV(ins_Copy(targetType), targetReg, srcReg, targetType);
     }
+}
+
+//----------------------------------------------------------------------
+// genCodeForBitCast - Generate code for a GT_BITCAST that is not contained
+//
+// Arguments
+//    treeNode - the GT_BITCAST for which we're generating code
+//
+void CodeGen::genCodeForBitCast(GenTreeOp* treeNode)
+{
+    regNumber targetReg  = treeNode->GetRegNum();
+    var_types targetType = treeNode->TypeGet();
+    GenTree*  op1        = treeNode->gtGetOp1();
+    genConsumeRegs(op1);
+
+    if (op1->isContained())
+    {
+        assert(op1->IsLocal() || op1->isIndir());
+        if (genIsRegCandidateLocal(op1))
+        {
+            unsigned lclNum = op1->AsLclVar()->GetLclNum();
+            GetEmitter()->emitIns_R_S(ins_Load(treeNode->TypeGet(), compiler->isSIMDTypeLocalAligned(lclNum)),
+                                      emitTypeSize(treeNode), targetReg, lclNum, 0);
+        }
+        else
+        {
+            op1->gtType = treeNode->TypeGet();
+            op1->SetRegNum(targetReg);
+            op1->ClearContained();
+            JITDUMP("Changing type of BITCAST source to load directly.");
+            genCodeForTreeNode(op1);
+        }
+    }
+    else
+    {
+        genBitCast(targetType, targetReg, op1->TypeGet(), op1->GetRegNum());
+    }
+    genProduceReg(treeNode);
 }
 
 //-------------------------------------------------------------------------- //

